pkg/cmd/server/origin/master_config.go

```diff
@@ -684,9 +684,9 @@ func newAuthenticator(config configapi.MasterConfig, restOptionsGetter restoptio
 			config.AuthConfig.RequestHeader.ExtraHeaderPrefixes,
 		)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("Error building front proxy auth config: %v", err)
 		}
-		ret.Handlers = append([]authenticator.Request{requestHeaderAuthenticator}, ret.Handlers...)
+		ret = unionrequest.NewUnionAuthentication(requestHeaderAuthenticator, ret)
```

Comment:
Give us a comment here.  I think this is saying, "you either use the front proxy authentication or you drop through to the 'normal' authentication chain we had before as a singular chunk".